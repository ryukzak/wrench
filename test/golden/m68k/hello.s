    .data
hello:           .byte  31, 'Hello\n\0World!\0\0\0' ; The string to print with newline and null terminators
output_addr:     .word  0x84               ; Output device address

    .text
    .org     0x100
_start:
    movea.l  output_addr, A1                 ; A1 <- address of output device address
    movea.l  (A1), A1                        ; A1 <- value at output_addr (0x84)

    move.l   hello, D0
    move.l   14, D1

loop:
    movea.l  D0, A0
    move.l   (A0), D2                        ; TODO: support for .b
    and.l    0xFF, D2
    move.l   D2, (A1)

    add.l    1, D0
    add.l    -1, D1
    bne      loop

    halt
