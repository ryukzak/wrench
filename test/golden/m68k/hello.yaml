name: "assert hello(*()) == ('Hello\n\0World',)"
limit: 1000
memory_size: 0x1000
input_streams:
  0x80: []
  0x84: []
reports:
  - name: Check results
    slice: last
    view: |
      numio[0x80]: {io:0x80:dec}
      numio[0x84]: {io:0x84:dec}
      symio[0x80]: {io:0x80:sym}
      symio[0x84]: {io:0x84:sym}
    assert: |
      numio[0x80]: [] >>> []
      numio[0x84]: [] >>> [72,101,108,108,111,10,0,87,111,114,108,100]
      symio[0x80]: "" >>> ""
      symio[0x84]: "" >>> "Hello\n\0World"
